lib/Album/AlbumMapper.php

```diff
@@ -0,0 +1,157 @@
+		$query = $this->connection->getQueryBuilder();
+		$query->select("name", "user")
+			->from("photos_albums")
+			->where($query->expr()->eq('album_id', $query->createNamedParameter($id, IQueryBuilder::PARAM_INT)));
+		$row = $query->executeQuery()->fetch();
+		if ($row) {
+			return new AlbumInfo($id, $row['user'], $row['name']);
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * @param string $userId
+	 * @return AlbumInfo[]
+	 */
+	public function getForUser(string $userId): array {
+		$query = $this->connection->getQueryBuilder();
+		$query->select("album_id", "name")
+			->from("photos_albums")
+			->where($query->expr()->eq('user', $query->createNamedParameter($userId)));
+		$rows = $query->executeQuery()->fetchAll();
+		return array_map(function (array $row) use ($userId) {
+			return new AlbumInfo($row['album_id'], $userId, $row['name']);
+		}, $rows);
+	}
+
+	public function rename(int $id, string $newName): void {
+		$query = $this->connection->getQueryBuilder();
+		$query->update("photos_albums")
+			->set("name", $query->createNamedParameter($newName))
+			->where($query->expr()->eq('album_id', $query->createNamedParameter($id, IQueryBuilder::PARAM_INT)));
+		$query->executeStatement();
+	}
+
+	public function delete(int $id): void {
+		$this->connection->beginTransaction();
+		$query = $this->connection->getQueryBuilder();
+		$query->delete("photos_albums")
+			->where($query->expr()->eq('album_id', $query->createNamedParameter($id, IQueryBuilder::PARAM_INT)));
+		$query->executeStatement();
+
+
+		$query = $this->connection->getQueryBuilder();
+		$query->delete("photos_albums_files")
+			->where($query->expr()->eq('album_id', $query->createNamedParameter($id, IQueryBuilder::PARAM_INT)));
+		$query->executeStatement();
+		$this->connection->commit();
+	}
+
+	/**
+	 * @param string $userId
+	 * @return AlbumWithFiles[]
+	 */
+	public function getForUserWithFiles(string $userId): array {
+		$query = $this->connection->getQueryBuilder();
+		$query->select("fileid", "mimetype", "a.album_id", "size", "mtime", "etag")
+			->selectAlias("f.name", "file_name")
+			->selectAlias("a.name", "album_name")
+			->from("photos_albums", "a")
+			->leftJoin("a", "photos_albums_files", "p", $query->expr()->eq("a.album_id", "p.album_id"))
+			->leftJoin("p", "filecache", "f", $query->expr()->eq("p.file_id", "f.fileid"))
+			->where($query->expr()->eq('user', $query->createNamedParameter($userId)));
+		$rows = $query->executeQuery()->fetchAll();
+
+		$filesByAlbum = [];
+		$albumsById = [];
+		foreach ($rows as $row) {
+			$albumId = $row['album_id'];
+			if ($row['fileid']) {
+				$mimeId = $row['mimetype'];
+				$mimeType = $this->mimeTypeLoader->getMimetypeById($mimeId);
+				$filesByAlbum[$albumId][] = new AlbumFile($row['fileid'], $row['file_name'], $mimeType, $row['size'], $row['mtime'], $row['etag']);
+			}
+
+			if (!isset($albumsById[$albumId])) {
+				$albumsById[$albumId] = new AlbumInfo($albumId, $userId, $row['album_name']);
```

Comment:
From my testing: `$albumId` is a string, but int is expected.

Reply:
Happens in other places

Reply:
should be fixed